db/migrate/20240304030429_topic_id_on_incoming_email_index.rb

```diff
@@ -2,7 +2,12 @@
 class TopicIdOnIncomingEmailIndex < ActiveRecord::Migration[7.0]
   disable_ddl_transaction!
 
-  def change
+  def up
+    remove_index :incoming_emails, :topic_id, if_exists: true
     add_index :incoming_emails, :topic_id, algorithm: :concurrently
   end
```

Comment:
```suggestion
  def up
    add_index :incoming_emails, :topic_id, algorithm: :concurrently, if_not_exists: true
  end
```

This is supposed to be equivalent I think

Reply:
I don't think that removes the index if it is invalid.

Reply:
oh yes right I misunderstood what you were trying to do here :+1:

Reply:
No worries at all. Like what @martin-brennan said in chat, we should probably introduce a helper method in core for this.